### PR TITLE
Added tests to the Math Magicians App

### PR DESCRIPTION
**Pull request by asifaminisonline**
Followed all instructions in the official documentation.
Added unit tests for the files operate.js and calculate.js using Jest.
Created unit tests for all React components.
Used Jest snapshots to test the components.
Used React Testing Library to simulate user interaction.
Run all full tests suite using Jest. All tests pass.
Followed best practices.
Followed Git flow.
Fix linter errors.

### CODE REVIEW

**Review by rloterh — COMMENTED** [review]
Hi Asif, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is not contributed by all (3) assigned pair group members. Kindly ensure that **all** assigned members to this project contribute evenly to this project, and then kindly request another code review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/tests/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  test('+/-', () => {
+    const obj = {
+      total: '50',
+      next: null,
+      operation: null,
+    };
+
+    const state = calculate(obj, '+/-');
+    expect(state.total).toBe('-50');
+  });
```
Comment (danmainah) [inline src/tests/calculate.test.js:55]: - [ ] @team great work on your tests however you haven't covered all scenarios, kindly check all scenario presented by the if statement and add test for them.

Comment (danmainah) [reply src/tests/calculate.test.js:55]: - [ ] @team kindly add a test for the app component , this is a project requirement 
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/tests/calculate.test.js (hunk 1 above)
Comment (asifaminisonline) [reply src/tests/calculate.test.js:55]: I am sorry, but if you will check again you can find all the tests inside the test folder which is inside src directory.

Comment (asifaminisonline) [reply src/tests/calculate.test.js:55]: Screenshot_2023-01-18-21-13-07-110_com android chrome
Screenshot_2023-01-18-21-13-39-206_com android chrome

Comment (asifaminisonline) [reply src/tests/calculate.test.js:55]: I have attached the images of test cases with folder directories.

Comment (danmainah) [reply src/tests/calculate.test.js:55]: @asifaminisonline I can't see the test for the App component, which I requested for review also for all scenarios in the calculate.js file

**Comment by asifaminisonline** [comment]
Error

**Comment by ahmetbozaci** [comment]
### Hello @asifaminisonline 
### _This is an additional review as a result of the second opinion request._ 

-  I see you create tests for the calculator file. The requirement doesn't say you should make tests for all scenarios. The tests that you create are enough to pass the project. But it will be good if you create more tests for other operations. OPTIONAL
- [ ]  According to [requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_tests.md#:~:text=Create%20unit%20tests%20for%20all%20React%20components%3A) you should create unit tests for all components. App.js file is also a component. Please create a unit test for this file. REQUIREMENT

_Please apply the requirement and then submit for review._



___



### Cheers and Happy coding! 💻 👏
____
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Pull request closed without merging**